src/Yaapii.Http/Wires/AspNetCore/AspNetCoreClient.cs

```diff
@@ -28,17 +28,21 @@
 namespace Yaapii.Http.Wires.AspNetCore
 {
     /// <summary>
-    /// Encapsulates a singleton <see cref="HttpClient"/>.
-    /// See https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/ for why this is necessary.
+    /// A <see cref="HttpClient"/> with the given timeout.
+    /// Clients will be reused, if the same timeout has been used before.
+    /// See https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/ for why reusing them is necessary.
+    /// Clients won't be reused if a previously unused timeout is needed, because the timeout can only be set before the first request is sent.
     /// </summary>
     public sealed class AspNetCoreClient : IScalar<HttpClient>
     {
-        private readonly static IList<HttpClient> client = new List<HttpClient>();
+        private readonly static IDictionary<long, HttpClient> clients = new Dictionary<long, HttpClient>();
```

Comment:
please do not use statics. They are forbidden in our rules. They are hard to test and bad for multithreading.
We must design our apps which need the client in a way that they all use the same object. 

You could design an interface IAspHttpClient to get a client with a timeout via a method 'Client(TimeSpan timeout)'. We can inject that as singleton into DI in Web.Asp dependent projects.